pkg/templates/templates_test.go

```diff
@@ -502,9 +517,44 @@ func TestResolveTemplate(t *testing.T) {
 	}
 
 	for _, test := range testcases {
-		tmplStr, _ := yamlToJSON([]byte(test.inputTmpl))
+		tmplStr, err := yamlToJSON([]byte(test.inputTmpl))
+		if err != nil {
+			t.Fatalf(err.Error())
+		}
+
+		// Disable encryption to prevent a panic
+		var enableEncryption, enableDecryption bool
+
+		if test.config.EncryptionConfig.EncryptionEnabled {
+			enableEncryption = true
+			test.config.EncryptionConfig.EncryptionEnabled = false
+		}
+
+		if test.config.EncryptionConfig.DecryptionEnabled {
+			enableDecryption = true
+			test.config.EncryptionConfig.DecryptionEnabled = false
+		}
+
 		resolver := getTemplateResolver(test.config)
 
+		// Restore and set encryption settings to be validated
+		test.config.EncryptionConfig.EncryptionEnabled = enableEncryption
+		test.config.EncryptionConfig.DecryptionEnabled = enableDecryption
+
+		err = resolver.SetEncryptionConfig(test.config.EncryptionConfig)
```

Comment:
This method is already tested in `TestSetEncryptionConfig`. Could you please remove the test cases that cause a panic in `getTemplateResolver`? Any missing test cases can be added to `TestNewResolver`.